### ISSUE TITLE
Strange behavior of <md-select> and overflow-y

### ISSUE THREAD

**Issue by datvm** [material-components/material-web#5195]
### What is affected?

Component

### Description

When `<md-select>` is inside a container with `overflow-y: auto`, the content would be clipped or a scrollbar is created:

![image](https://github.com/material-components/material-web/assets/6388546/eceb1dd6-55aa-4345-bcc2-47a76a399f48)

Just a bit related but the floating label of `<md-text-field>` has the same problem as well:

![image](https://github.com/material-components/material-web/assets/6388546/cf610359-d716-4de0-8770-0401119bcd14)

### Reproduction

[Lit Playground](https://lit.dev/playground/#project=W3sibmFtZSI6InNpbXBsZS1ncmVldGluZy50cyIsImNvbnRlbnQiOiJpbXBvcnQgXCJAbWF0ZXJpYWwvd2ViL2FsbFwiIn0seyJuYW1lIjoiaW5kZXguaHRtbCIsImNvbnRlbnQiOiI8IURPQ1RZUEUgaHRtbD5cbjxoZWFkPlxuICA8bGluayBocmVmPVwiaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9ib290c3RyYXBANS4zLjIvZGlzdC9jc3MvYm9vdHN0cmFwLm1pbi5jc3NcIiByZWw9XCJzdHlsZXNoZWV0XCIgaW50ZWdyaXR5PVwic2hhMzg0LVQzYzZDb0lpNnVMckE5VG5lTkVvYTdSeG5hdHpqY0RTQ21HMU1YeFNSMUdBc1hFVi9Ed3d5a2MyTVBLOE0ySE5cIiBjcm9zc29yaWdpbj1cImFub255bW91c1wiPlxuICA8c2NyaXB0IHR5cGU9XCJtb2R1bGVcIiBzcmM9XCIuL3NpbXBsZS1ncmVldGluZy5qc1wiPjwvc2NyaXB0PlxuPC9oZWFkPlxuPGJvZHkgc3R5bGU9XCJoZWlnaHQ6IDEwMHZoO1wiPlxuICA8ZGl2IHN0eWxlPVwiZGlzcGxheTogZmxleDsgZmxleC1kaXJlY3Rpb246IGNvbHVtbjsgaGVpZ2h0OiAxMDB2aDtcIj5cbiAgICAgIDxwPlNvbWUgY29udGVudDwvcD5cbiAgICAgIDxwPlNvbWUgY29udGVudDwvcD5cbiAgICAgIDxwPlNvbWUgY29udGVudDwvcD5cbiAgICAgIDxwPlNvbWUgY29udGVudDwvcD5cbiAgICAgIFxuICAgICAgPGRpdiBzdHlsZT1cImZsZXgtZ3JvdzogMTsgbWluLWhlaWdodDogMDsgb3ZlcmZsb3cteTogYXV0bztcIj4gICAgICBcbiAgICAgICAgPHA-U29tZSBjb250ZW50PC9wPlxuICAgICAgICA8cD5Tb21lIGNvbnRlbnQ8L3A-XG4gICAgICAgIFxuICAgICAgICA8ZGl2PlxuICAgICAgICAgIDxtZC1vdXRsaW5lZC1zZWxlY3Qgc3R5bGU9XCJ3aWR0aDogMTAwJTtcIj5cbiAgICAgICAgICAgIDxtZC1zZWxlY3Qtb3B0aW9uPjE8L21kLXNlbGVjdC1vcHRpb24-XG4gICAgICAgICAgICA8bWQtc2VsZWN0LW9wdGlvbj4yPC9tZC1zZWxlY3Qtb3B0aW9uPlxuICAgICAgICAgICAgPG1kLXNlbGVjdC1vcHRpb24-MzwvbWQtc2VsZWN0LW9wdGlvbj5cbiAgICAgICAgICAgIDxtZC1zZWxlY3Qtb3B0aW9uPjQ8L21kLXNlbGVjdC1vcHRpb24-ICAgICAgXG4gICAgICAgICAgPC9tZC1vdXRsaW5lZC1zZWxlY3Q-ICBcbiAgICAgICAgPC9kaXY-ICAgICAgICBcbiAgICAgICAgXG4gICAgICAgIDxwPlNvbWUgY29udGVudDwvcD5cbiAgICAgICAgPHA-U29tZSBjb250ZW50PC9wPlxuICAgICAgPC9kaXY-XG4gICAgICA8cD5Tb21lIGNvbnRlbnQ8L3A-XG4gICAgICA8cD5Tb21lIGNvbnRlbnQ8L3A-XG4gICAgICA8cD5Tb21lIGNvbnRlbnQ8L3A-XG4gICAgICA8cD5Tb21lIGNvbnRlbnQ8L3A-XG4gICAgICA8cD5Tb21lIGNvbnRlbnQ8L3A-XG4gICAgICA8cD5Tb21lIGNvbnRlbnQ8L3A-XG4gICAgICA8cD5Tb21lIGNvbnRlbnQ8L3A-XG4gICAgICA8cD5Tb21lIGNvbnRlbnQ8L3A-XG4gIDwvZGl2PiAgICBcbiAgPC9kaXY-ICBcbjwvYm9keT5cbiJ9LHsibmFtZSI6InBhY2thZ2UuanNvbiIsImNvbnRlbnQiOiJ7XG4gIFwiZGVwZW5kZW5jaWVzXCI6IHtcbiAgICBcImxpdFwiOiBcIl4zLjAuMFwiLFxuICAgIFwiQGxpdC9yZWFjdGl2ZS1lbGVtZW50XCI6IFwiXjIuMC4wXCIsXG4gICAgXCJsaXQtZWxlbWVudFwiOiBcIl40LjAuMFwiLFxuICAgIFwibGl0LWh0bWxcIjogXCJeMy4wLjBcIlxuICB9XG59IiwiaGlkZGVuIjp0cnVlfV0)

![image](https://github.com/material-components/material-web/assets/6388546/a19bdcf3-6cf1-4701-abe6-27898a7506b4)

Basically I want my main content to be "sandwiched" between two horizontal content and it may have a scrollbar when needed but right now the menu is clipped instead. On Lit Playground there is a scrollbar but in my WebView app the scrollbar doesn't appear at all and user cannot see all the options.

### Workaround

None.

### Is this a regression?

Yes. This used to work, but now it doesn't.

### Affected versions

1.0.1

### Browser/OS/Node environment

Tested on Edge, Chrome and Firefox. They all have the same behavior.

**Comment by asyncLiz** [comment]
Try adding `menu-positioning="fixed"` to the select. This is mentioned in the [API docs](https://github.com/material-components/material-web/blob/main/docs/components/select.md#api):

> Whether or not the underlying md-menu should be position: fixed to display in a top-level manner, or position: absolute.
position:fixed is useful for cases where select is inside of another element with stacking context and hidden overflows such as md-dialog.

**Comment by asyncLiz** [comment]
> Just a bit related but the floating label of <md-text-field> has the same problem as well

If an outlined field is against the edge of an overflow hidden surface, you'll need to add margin to let the label show. You can also add a default top margin to all outlined selects/text fields to reserve space for their floating labels.

**Comment by datvm** [comment]
Ah you are right, I have worked with that one before but somehow I forgot about it. Maybe I thought it only applies to `<md-menu>`. Thanks!
